asynq/async_task.py

```diff
@@ -357,18 +359,18 @@ def dump(self, indent=0):
     def _enter_context(self, context):
         if _debug_options.DUMP_CONTEXTS:
             debug.write('@async: +context: %s' % debug.str(context))
-        self._contexts.append(context)
+        self._contexts[id(context)] = context
 
     def _leave_context(self, context):
         if _debug_options.DUMP_CONTEXTS:
             debug.write('@async: -context: %s' % debug.str(context))
-        self._contexts.remove(context)
+        del self._contexts[id(context)]
 
     def _pause_contexts(self):
         if not self._contexts_active:
             return
         self._contexts_active = False
-        contexts = self._contexts
+        contexts = self._contexts.values()
```

Comment:
This will break line 380/382 below; you can't index an odict_values object. You'll have to `list()` this.